### DISCUSSION TITLE
API Rule returns item when checking if field matches header when it shouldn't?

### DISCUSSION THREAD

**Discussion by EverlyScott** [pocketbase/pocketbase#4695] [Q&A]
Hi,

I'm trying to only allow users to access a group's info if they have the `joinCode` of the group, so when making the request to view the group I'm putting a `Join-Code` value in the request header so pb can only display the group with that code. Here's the request I'm making
```bash
curl 'http://127.0.0.1:8090/api/collections/groups/records?page=1&perPage=1&skipTotal=1' \
  -H 'Authorization: abcd1234thisisatoken' \
  -H 'Join-Code: 12345678'
```
And here's my API Rule for listing and viewing:
```
@request.auth.id != "" && (@request.auth.joinedGroups.id ?= id || @request.headers.join_code = joinCode)
```

If I make the `Join-Code` header match a group's joinCode field, it works properly, but if I make the header not match any group, I still get the group with the oldest `created` date, even though it doesn't match the join code. Is this intended? If so how can I do this properly?

**Comment by ganigeorgiev** [comment]
I'm confused and not sure that I understand what actually is your question. The filter has no relation to how the records are sorted in the return result. If you want to allow access **only** when the header is present then change the `||` with `&&`.

If you need further help please export your collections structure as json (_Admin UI > Settings > Export collections_).
